### PR TITLE
Add onClick behaviour to links to new tooltips

### PR DESCRIPTION
**Pull request by javiruiz01**
**What is this feature?**

This PR adds a call to the `onClick` function of a link if it exists. It replicates the behaviour of the previous tooltips (it was using the `DataLink` component, and handling the link's `onClick` ([ref](https://github.com/grafana/grafana/blob/b20da139ad0fe96299321c347baa4da36860ea54/packages/grafana-ui/src/components/DataLinks/DataLinkButton.tsx#L22)))

**Why do we need this feature?**

We need this feature to be able to override the link and behaviour when clicking on it, since we usually have the information related to that point in there.

**Who is this feature for?**

It replicates existing behaviour in the old tooltips, so for everyone relying on the `onClick` attribute of the link.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/visualization/data-hover/ExemplarHoverView.tsx [hunk 1]
```diff
@@ -42,7 +42,21 @@ export const ExemplarHoverView = ({ displayValues, links, header = 'Exemplar' }:
...
+                link.onClick
+                  ? (event) => {
+                      if (!(event.ctrlKey || event.metaKey || event.shiftKey) && link.onClick) {
+                        event.preventDefault();
+                        link.onClick(event);
+                      }
+                    }
+                  : undefined
+              }
+            >
```
Comment (adela-almasan) [inline public/app/features/visualization/data-hover/ExemplarHoverView.tsx:59]:
```suggestion
<DataLinkButton link={link} key={i} buttonProps={{ size: 'md' }} />
```

Comment (adela-almasan) [reply public/app/features/visualization/data-hover/ExemplarHoverView.tsx:59]: @javiruiz01 I think we can just reuse `DataLinkButton` and cleanup the file and we should be good to merge!

**Review by codeincarnate — APPROVED** [review]
Adela's comment is on point but otherwise looks good to me!

File: public/app/features/visualization/data-hover/ExemplarHoverView.tsx (hunk 1 above)
Comment (javiruiz01) [reply public/app/features/visualization/data-hover/ExemplarHoverView.tsx]: Oh that's so much better, I updated it, thanks!

**Review by adela-almasan — APPROVED** [review]

**Pull request merged**